atom/browser/api/atom_api_app.cc

```diff
@@ -268,6 +269,30 @@ v8::Local<v8::Value> App::DefaultSession(v8::Isolate* isolate) {
     return v8::Local<v8::Value>::New(isolate, default_session_);
 }
 
+bool App::MakeSingleInstance(v8::Local<v8::Function> callback) {
```

Comment:
You can pass `const ProcessSingleton::NotificationCallback& callback` instead of `v8::Function` here. The conversion magic will happen automatically.